app/src/main/java/eu/kanade/tachiyomi/ui/library/LibraryController.kt

```diff
@@ -1751,12 +1750,62 @@ class LibraryController(
                 presenter.downloadUnread(selectedMangas.toList())
             }
             R.id.action_mark_as_read -> {
-                presenter.markReadStatus(selectedMangas.toList(), true)
+                val markRead = true
```

Comment:
im not in full love with the undo snackbar for this but eh its fine.

Instead, can you keep it, but also add a dialog for confirmation for mark as read/unread? like the one in manga details, and then show this undo snackbar.

Reply:
Done
The main point of the snackbar was to allow rollback in case of unwanted click. The confirmation dialog prevents unwanted click so the snackbar is not mandatory anymore, if you really don't like it I can remove it